app/controlplane/pkg/biz/organization.go

```diff
@@ -244,3 +264,75 @@ func (uc *OrganizationUseCase) Delete(ctx context.Context, id string) error {
 	// Delete the organization
 	return uc.orgRepo.Delete(ctx, orgUUID)
 }
+
+// AutoOnboardOrganizations creates the organizations specified in the onboarding config and assigns the user to them
+// with the specified role if they are not already a member.
+func (uc *OrganizationUseCase) AutoOnboardOrganizations(ctx context.Context, userID string) error {
+	// Parse user UUID
+	userUUID, err := uuid.Parse(userID)
+	if err != nil {
+		return NewErrInvalidUUID(err)
+	}
+
+	for _, spec := range uc.onboardingConfig {
+		// Ensure the organization exists or create it if it doesn't
+		org, err := uc.ensureOrganizationExists(ctx, spec)
+		if err != nil {
+			return fmt.Errorf("failed to ensure organization exists: %w", err)
+		}
+
+		// Parse organization UUID
+		orgUUID, err := uuid.Parse(org.ID)
+		if err != nil {
+			return NewErrInvalidUUID(err)
+		}
+
+		// Ensure user membership
+		if err := uc.ensureUserMembership(ctx, orgUUID, userUUID, spec); err != nil {
+			return fmt.Errorf("failed to ensure user membership: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// ensureOrganizationExists ensures that an organization specified by the onboarding configuration exists.
+// If the organization does not exist, it creates it.
+func (uc *OrganizationUseCase) ensureOrganizationExists(ctx context.Context, spec *conf.OnboardingSpec) (*Organization, error) {
+	// Ensure the organization exists or create it if it doesn't
+	org, err := uc.FindByName(ctx, spec.GetName())
+	if err != nil {
+		if errors.As(err, &ErrNotFound{}) {
+			org, err = uc.Create(ctx, spec.GetName(), WithCreateInlineBackend())
+			if err != nil {
+				return nil, fmt.Errorf("failed to create organization: %w", err)
+			}
+		} else {
+			return nil, fmt.Errorf("failed to find organization: %w", err)
+		}
+	}
+
+	return org, nil
+}
+
+// ensureUserMembership ensures that a user is a member of the specified organization with the appropriate role.
+// If the membership does not exist, it creates it.
+func (uc *OrganizationUseCase) ensureUserMembership(ctx context.Context, orgUUID, userUUID uuid.UUID, spec *conf.OnboardingSpec) error {
+	m, err := uc.membershipRepo.FindByOrgAndUser(ctx, orgUUID, userUUID)
+	if err != nil && !errors.As(err, &ErrNotFound{}) {
```

Comment:
are you sure this error handling is correct? I think that method returns `m=nil` if not found, not ErrNotFound?